### DISCUSSION TITLE
Rebinning in the framework

### DISCUSSION THREAD

**Discussion by Tomoya-Iizawa** [scikit-hep/cabinetry#451] [Q&A] [answered]
Hi,
Is there any way to do rebinning in the framework?
For example, if original histograms have bin edges [0, 250, 500, 750, 1000], is it possible to run fit for histograms of bin edges [0, 500, 1000]? If yes, is it possible to define binning in each region? e.g. use binning [0, 250, 500, 750, 1000] for 0bjet region, while [0, 500, 1000] for 1b jet region.

I tried to set by "Binning" setup in "Regions" category, as
  - Name: "HighMassOSs2thh_1bjets"
    RegionPath: "rec_mll_1bjets_HighMassOSs2thh"
    Binning: [0, 500, 1000]
    Variable: $m_{\ell\ell}$ [GeV]
but the outcome did not change compared with original binning, so I do not think this works.

**Comment by alexander-held** [comment, marked as answer]
Not currently, but this is planned: see #412 and a draft in #418. The best way to support this in my opinion is along the lines of what you show, allowing users provide explicit bin edges. This requires the feature to be supported upstream in `hist`, see https://github.com/scikit-hep/hist/issues/345. At the moment this upstream issue blocks a proper implementation. The branch in #418 only supports uniform rebinning right now (which works for your example, but would e.g. not work for a target binning of [0, 250, 1000]).

An example for how to currently achieve rebinning can be found here: [high-level API](https://github.com/iris-hep/analysis-grand-challenge/blob/4497193659206089323d948f2ae315257a976897/analyses/cms-open-data-ttbar/ttbar_analysis_pipeline.py#L524-L525) and [implementation](https://github.com/iris-hep/analysis-grand-challenge/blob/4497193659206089323d948f2ae315257a976897/analyses/cms-open-data-ttbar/utils/__init__.py#L174-L193). It makes use of [template builder overrides](https://cabinetry.readthedocs.io/en/latest/advanced.html#overrides-for-template-building) and works with histogram inputs. You would have to hardcode the new binning in there and manually re-bin (you can use config information to make decisions in that function, as this information is exposed).

**Reply by Tomoya-Iizawa** [reply]
I see, thank you! hm, I will try that, or try to prepare histograms of different binning among regions.
